### PR TITLE
(DOCSP-20059): Add performance optimizations and limitations for Realm Triggers

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Add sections about performance optimization and limitations per TSE feedback. mostly consolidation of existing content that was spread across easy-to-miss notes. also refactored the Triggers intro page to be more general and include the triggers limitations section. 

### Jira

- https://jira.mongodb.org/browse/DOCSP-20059

### Staged Changes (Requires MongoDB Corp SSO)

- [About Triggers](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20059/triggers/database-triggers/)
- [Database Triggers](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20059/triggers/database-triggers/)
- [Scheduled Triggers](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20059/triggers/scheduled-triggers/) 
- [Send Triggers to AWS EventBridge](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20059/triggers/examples/send-events-aws-eventbridge/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/triggers/database-triggers.txt [hunk 1]
```diff
@@ -351,9 +338,16 @@ When :ref:`resuming the suspended Trigger <resume-a-suspended-trigger>`,
...
-new events but does not process any missed past events. You can adjust
-your oplog size to keep the resume token around for more time after a
-suspension.
+new events but does not process any missed past events. 
+
+You can adjust the oplog size to keep the resume token for more time after a suspension
+by :atlas:`scaling your {+atlas-short+} cluster </scale-cluster/>`. 
+Maintain an oplog size a few times greater than
+your cluster's peak oplog throughput (GB/hour) to reduce the risk of a 
+suspended trigger's resume token from dropping off the oplog
```
Comment (cbush) [inline source/triggers/database-triggers.txt:347]:
```suggestion
suspended trigger's resume token dropping off the oplog 
```

File: source/triggers/database-triggers.txt [hunk 2]
```diff
@@ -351,9 +338,16 @@ When :ref:`resuming the suspended Trigger <resume-a-suspended-trigger>`,
...
-your oplog size to keep the resume token around for more time after a
-suspension.
+new events but does not process any missed past events. 
+
+You can adjust the oplog size to keep the resume token for more time after a suspension
+by :atlas:`scaling your {+atlas-short+} cluster </scale-cluster/>`. 
+Maintain an oplog size a few times greater than
+your cluster's peak oplog throughput (GB/hour) to reduce the risk of a 
+suspended trigger's resume token from dropping off the oplog 
+before its execution.
```
Comment (cbush) [inline source/triggers/database-triggers.txt:348]: "Its" refers to "resume token":
```suggestion
before the trigger executes. 
```

File: source/triggers/database-triggers.txt [hunk 3]
```diff
@@ -379,3 +373,108 @@ mitigations:
...
+significantly faster (depending on your use case) but does not guarantee that
+multiple executions of a Trigger Function occur in event order.
+
+.. _database-triggers-disable-collection-level-preimages:
+
+Disable Collection-Level Preimages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+          
+Document preimages use extra information stored in the oplog.
+The extra data may have performance implications for some apps.
```
Comment (cbush) [inline source/triggers/database-triggers.txt:406]: What performance implications, what apps? This is vague.

File: source/triggers/database-triggers.txt [hunk 4]
```diff
@@ -379,3 +373,108 @@ mitigations:
...
+Once you've enabled document preimages for any trigger on a
+given collection, that collection will include preimage data in
+the oplog and other triggers on the collection can use preimages
+with no additonal overhead.
+
+You can disable document preimages on a per-trigger basis to
+exclude the preimage from change events. Regardless of your
+trigger-level settings, a collection's oplog entries will
+continue to include preimage data unless you explicitly disable
+preimages for the collection.
```
Comment (cbush) [inline source/triggers/database-triggers.txt:417]: Cool, but how do we access that preimage in the trigger function?

**Review by cbush — APPROVED** [review]
Great!

File: source/triggers/database-triggers.txt (hunk 4 above)
Comment (mongodben) [reply source/triggers/database-triggers.txt:417]: this is included in lines 220 - 223, in addition to the above examples. 

the text here is just about a performance optimization

File: source/triggers/database-triggers.txt (hunk 3 above)
Comment (mongodben) [reply source/triggers/database-triggers.txt:406]: specified that it can degrade performance

**Pull request merged**